exit.go

```diff
@@ -4,127 +4,165 @@ import (
+		hasVolume bool
+	)
+	hasVolume, err = dckr.VolumeExists(id)
 	if err != nil {
-		logcabin.Error.Print(err)
-		inputContainers = []string{}
+		return errors.Wrap(err, "failed to check for volume existence")
 	}
-	for _, ic := range inputContainers {
-		logcabin.Info.Printf("Nuking input container %s", ic)
-		err = dckr.NukeContainer(ic)
-		if err != nil {
-			logcabin.Error.Print(err)
+	if hasVolume {
+		log.Infof("removing volume: %s", id)
+		if err = dckr.RemoveVolume(id); err != nil {
+			return errors.Wrap(err, "failed to remove volume")
 		}
 	}
+	return nil
+}
 
-	logcabin.Info.Println("Finding all step containers")
-	stepContainers, err := dckr.ContainersWithLabel(dockerops.TypeLabel, strconv.Itoa(dockerops.StepContainer), true)
+// RemoveJobContainers removes containers based on their job label.
+func RemoveJobContainers(id string) error {
+	log.Infof("Finding all containers with the label %s=%s", model.DockerLabelKey, id)
+	jobContainers, err := dckr.ContainersWithLabel(model.DockerLabelKey, id, true)
 	if err != nil {
-		logcabin.Error.Print(err)
+		return errors.Wrapf(err, "failed to find containers with label %s=%s", model.DockerLabelKey, id)
 	}
-	for _, sc := range stepContainers {
-		logcabin.Info.Printf("Nuking step container %s", sc)
-		err = dckr.NukeContainer(sc)
+	for _, jc := range jobContainers {
+		log.Infof("Nuking container %s", jc)
+		err = dckr.NukeContainer(jc)
 		if err != nil {
-			logcabin.Error.Print(err)
+			return errors.Wrapf(err, "failed to remove container %s", jc)
 		}
 	}
+	return nil
+}
 
-	logcabin.Info.Println("Finding all data containers")
+// RemoveDataContainers attempts to remove all data containers.
+func RemoveDataContainers() error {
+	log.Infoln("Finding all data containers")
 	dataContainers, err := dckr.ContainersWithLabel(dockerops.TypeLabel, strconv.Itoa(dockerops.DataContainer), true)
 	if err != nil {
-		logcabin.Error.Print(err)
+		return errors.Wrapf(err, "failed to find containers with label %s=%s", dockerops.TypeLabel, strconv.Itoa(dockerops.DataContainer))
 	}
 	for _, dc := range dataContainers {
-		logcabin.Info.Printf("Nuking data container %s", dc)
+		log.Infof("Nuking data container %s", dc)
 		err = dckr.NukeContainer(dc)
 		if err != nil {
-			logcabin.Error.Print(err)
+			return errors.Wrapf(err, "failed to remove container %s", dc)
+		}
+	}
+	return nil
+}
+
+// RemoveStepContainers attempts to remove all step containers.
+func RemoveStepContainers() error {
+	log.Infoln("Finding all step containers")
+	stepContainers, err := dckr.ContainersWithLabel(dockerops.TypeLabel, strconv.Itoa(dockerops.StepContainer), true)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find containers with label %s=%s", dockerops.TypeLabel, strconv.Itoa(dockerops.StepContainer))
+	}
+	for _, sc := range stepContainers {
+		log.Infof("Nuking step container %s", sc)
+		err = dckr.NukeContainer(sc)
+		if err != nil {
+			return errors.Wrapf(err, "failed to remove container %s", sc)
```

Comment:
This is another case where the best bet seems to be to log the error and move on.